sbi/src/api/attestation.rs

```diff
@@ -13,21 +14,134 @@ pub const MAX_CERT_SIZE: usize = 4096;
 /// The data blob passed to the GetEvidence call must be 64 bytes long.
 pub const REQUEST_DATA_SIZE: usize = 64;
 
-/// Attestation evidence formats.
-#[derive(Copy, Clone, Debug)]
-#[repr(u64)]
-pub enum EvidenceFormat {
-    /// Single layer DICE TCB
-    /// https://trustedcomputinggroup.org/resource/dice-attestation-architecture/
-    DiceTcbInfo = 0,
-
-    /// X.509 extension for multiple layers DICE TCB.
-    /// https://trustedcomputinggroup.org/resource/dice-attestation-architecture/
-    DiceMultiTcbInfo = 1,
-
-    /// Open DICE profile
-    /// https://pigweed.googlesource.com/open-dice/+/HEAD/docs/specification.md
-    OpenDice = 2,
+/// A list of supported hash algorithms.
+#[derive(Copy, Clone, Default, Debug)]
+#[repr(u8)]
+pub enum HashAlgorithm {
+    /// SHA-384
+    #[default]
+    Sha384 = 1,
+    /// SHA-512
+    Sha512 = 2,
+}
+
+flags! {
+    /// Attestation evidence formats.
+    #[derive(Default)]
+    #[repr(u8)]
+    pub enum EvidenceFormat: u8 {
+        /// Single layer DICE TCB
+        /// https://trustedcomputinggroup.org/resource/dice-attestation-architecture/
+        #[default]
+        DiceTcbInfo = 1,
+
+        /// X.509 extension for multiple layers DICE TCB.
+        /// https://trustedcomputinggroup.org/resource/dice-attestation-architecture/
+        DiceMultiTcbInfo = 2,
+
+        /// Open DICE profile
+        /// https://pigweed.googlesource.com/open-dice/+/HEAD/docs/specification.md
+        OpenDice = 4,
+    }
+}
+
+/// Attestation Capabilities
+///
+/// This structure exposes the supported attestation capabilities to the SBI
+/// GetCapabilities caller. It lets the caller know which hash algorithms,
+/// evidence formats, and measurements mappings the SBI implementation supports.
+#[repr(C)]
+#[derive(Default)]
+pub struct AttestationCapabilities {
+    /// The TCB Secure Version Number.
+    pub tcb_svn: u64,
+    /// The supported hash algorithm.
+    pub hash_algorithm: HashAlgorithm,
+    /// The supported evidence formats. This is a bitmap.
+    pub evidence_formats: FlagSet<EvidenceFormat>,
+    /// Number of static measurement registers.
+    pub static_measurements: u8,
+    /// Number of runtime measurement registers.
+    pub runtime_measurements: u8,
+    /// Array of all measurement register descriptors.
+    pub measurement_registers: [MeasurementRegisterDescriptor; 32],
+
```

Comment:
Nit: Delete newline or add a comment? Also, seems off to have an one-off non-public member, but it may be necessary in this case.

Reply:
I added a comment. Yes, this is an internal field for the measurement array index tracking. We check for out of bounds and return an error when that happens.